utils/polus-bfio-util/bfio/bfio.py

```diff
@@ -470,152 +223,309 @@ def _fetch(self):
+                                       X[ind][0] - self._tile_x_offset:X[ind][1] - self._tile_x_offset]
+
         return images
-    
-    def maximum_batch_size(self,tile_size,tile_stride=None):
-        """maximum_batch_size Maximum allowable batch size for tiling
-        The pixel buffer only loads at most two supertiles at a time. If the batch
-        size is too large, then the tiling function will attempt to create more
-        tiles than what the buffer holds. To prevent the tiling function from doing
-        this, there is a limit on the number of tiles that can be retrieved in a
-        single call. This function determines what the largest number of retreivable
-        batches is.
+
+    def __call__(self,
+                 tile_size: typing.Union[list,tuple],
+                 tile_stride: typing.Union[list,tuple,None] = None,
+                 batch_size: typing.Union[int,None] = None,
+                 channels: typing.Union[list] = [0]) -> typing.Iterable[typing.Tuple[numpy.ndarray,tuple]]:
+        """Iterate through tiles of an image
+        
+        The BioReader object can be called, and will act as an iterator to load
+        tiles of an image. The iterator buffers the loading of pixels
+        asynchronously to quickly deliver images of the appropriate size.
+        
         Args:
-            tile_size (list): The height and width of the tiles to retrieve
-            tile_stride (list, optional): If None, defaults to tile_size.
-                Defaults to None.
+            tile_size: A list/tuple of length 2, indicating the height and width
+                of the tiles to return.
+            tile_stride: A list/tuple of length 2, indicating the row and column
+                stride size. If None, then tile_stride = tile_size. *Defaults to
+                None.*
+            batch_size: Number of tiles to return on each iteration. *Defaults
+                to None, which is the smaller of 32 or the*
+                :attr:`~.maximum_batch_size`
+            channels: A placeholder. Only the first channel
+                is ever loaded. *Defaults to [0].*
+                
         Returns:
-            int: Maximum allowed number of batches that can be retrieved by the
-                iterate method.
-        """
-        if tile_stride == None:
-            tile_stride = tile_size
+            A tuple containing a 4-d numpy array and a tuple containing a list
+            of X,Y,Z,C,T indices. The numpy array has dimensions 
+            ``[tile_num,tile_size[0],tile_size[1],channels]``
+
+        Example:
+            .. code:: python
             
-        xyoffset = [(tile_size[0]-tile_stride[0])/2,(tile_size[1]-tile_stride[1])/2]
+                from bfio import BioReader
+                import matplotlib.pyplot as plt
+
+                br = BioReader('/path/to/file')
+
+                for tiles,ind in br(tile_size=[256,256],tile_stride=[200,200]):
```

Comment:
I tend to find it more intuitive to return the indices then the data. So the order reversed: `ind, tiles`. Seems to be the operative paradigm across code.